hydroflow_lang/src/graph/ops/demux.rs

```diff
@@ -0,0 +1,89 @@
+use crate::graph::PortIndexValue;
+
+use super::{
+    OperatorConstraints, OperatorWriteOutput, WriteContextArgs, WriteIteratorArgs, RANGE_1,
+};
+
+use quote::{quote_spanned, ToTokens};
+use syn::spanned::Spanned;
+use syn::{parse_quote, ExprClosure};
+
+// TODO(mingwei): Preprocess rustdoc links in mdbook or in the `operator_docgen` macro.
+
+/// > Arguments: A Rust closure, the first argument is a received item and the
+/// > second argument is a [`tl!` tuple list](https://hydro-project.github.io/hydroflow/doc/hydroflow/macro.tl.html)
+/// > where each item name is an output port.
+///
+/// Takes the input stream and allows the user to determine what elemnt(s) to
+/// deliver to any number of output streams.
+///
+/// > Note: Downstream operators may need explicit type annotations.
+///
+/// > Note: Import the [`Pusherator`](https://hydro-project.github.io/hydroflow/doc/pusherator/trait.Pusherator.html)
+/// > trait to use the [`.give(...)` method](https://hydro-project.github.io/hydroflow/doc/pusherator/trait.Pusherator.html#tymethod.give).
+///
+/// ```hydroflow
+/// demux = recv_iter(1..=100) -> demux(|v, tl!(fzbz, fizz, buzz, vals)|
```

Comment:
Confusing to use variable names that are the same as keywords, even if it's legal.